lms/djangoapps/course_api/blocks/transformers/milestones.py

```diff
@@ -76,3 +103,50 @@ def has_pending_milestones_for_user(block_key, usage_info):
             'requires',
             usage_info.user.id
         ))
+
+    def add_special_exam_info(self, block_key, block_structure, usage_info):
+        """
+        Adds special exam information to course blocks.
+        """
+        if self.is_special_exam(block_key, block_structure):
+
+            # call into edx_proctoring subsystem to get relevant special exam information
+            #
+            # This will return None, if (user, course_id, content_id) is not applicable
+            special_exam_attempt_context = None
+            try:
+                special_exam_attempt_context = get_attempt_status_summary(
+                    usage_info.user.id,
+                    unicode(block_key.course_key),
+                    unicode(block_key)
+                )
+            except ProctoredExamNotFoundException as ex:
+                log.exception(ex)
+
+            if special_exam_attempt_context:
```

Comment:
Nit: For readability and simpler code, put this in an `else` clause of the `try`-`except`.  And you can also eliminate the `special_exam_attempt_context` boolean,